### PR TITLE
GH2530&2531: Rewrite in-process NuGet and update to 5.0.0

### PR DESCRIPTION
**Pull request by mholo65**
Rewrite in-process NuGet to get rid of NuGet.PackageManagement dependency and update to NuGet Client libs to 5.0.0.

### CODE REVIEW

**Comment by mholo65** [comment]
Fixes #2530 
Fixes #2531

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettingSection.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace Cake.NuGet.Tests.Stubs
+{
+    internal sealed class FakeNuGetSettingSection : SettingSection
+    {
+        public FakeNuGetSettingSection(string name, IReadOnlyDictionary<string, string> attributes, IEnumerable<SettingItem> children)
+            : base(name, attributes, children)
+        {
+        }
+
+        public void AddItem(SettingItem item) => Children.Add(item);
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettingSection.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettingSection.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+        public override SettingBase Clone() => new FakeNuGetSettingSection(ElementName, MutableAttributes, Items);
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettingSection.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+using System.Linq;
+using NuGet.Configuration;
+
+namespace Cake.NuGet.Tests.Stubs
+{
+    internal sealed class FakeNuGetSettings : ISettings
+    {
+        public event EventHandler SettingsChanged;
+
+        private IDictionary<string, FakeNuGetSettingSection> _settings = new Dictionary<string, FakeNuGetSettingSection>(StringComparer.OrdinalIgnoreCase);
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs]: Initialize fields in the constructor

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                section = new FakeNuGetSettingSection(sectionName, null, Enumerable.Empty<SettingItem>());
+                _settings[sectionName] = section;
+            }
+
+            section.AddItem(item);
+
+            SettingsChanged?.Invoke(this, new EventArgs());
+        }
+
+        public IList<string> GetConfigFilePaths() => Array.Empty<string>();
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+        public IList<string> GetConfigRoots() => Array.Empty<string>();
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

File: src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+        public SettingSection GetSection(string sectionName) =>
```
Comment (patriksvensson) [inline src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

File: src/Cake.NuGet/Install/NuGetPackageInstaller.cs [hunk 7]
```diff
@@ -35,9 +35,16 @@ internal sealed class NuGetPackageInstaller : INuGetPackageInstaller, IDisposabl
         private readonly ISettings _nugetSettings;
         private readonly ILogger _nugetLogger;
         private readonly NuGetFramework _currentFramework;
-        private readonly GatherCache _gatherCache;
         private readonly SourceCacheContext _sourceCacheContext;
 
+        private static readonly ISet<string> _blackListedPackages = new HashSet<string>(new[]
```
Comment (patriksvensson) [inline src/Cake.NuGet/Install/NuGetPackageInstaller.cs]: Move assignment to static constructor

File: src/Cake.NuGet/Install/NuGetSourceRepositoryProvider.cs [hunk 8]
```diff
@@ -105,28 +112,16 @@ SourceRepository GetOrCreateRepository(string source)
...
-        {
-            var repository = new SourceRepository(source, _resourceProviders, type);
-            var comparer = new SourceRepositoryComparer();
+        public IEnumerable<SourceRepository> Repositories => _repositories;
 
-            if (_repositories.Contains(repository))
-            {
-                return _repositories.First(x => comparer.Equals(x, repository));
-            }
+        private SourceRepository CreateRepository(string source) => CreateRepository(new PackageSource(source));
```
Comment (patriksvensson) [inline src/Cake.NuGet/Install/NuGetSourceRepositoryProvider.cs]: Don't use expression-bodied methods. We're not using it anywhere else in the code base.

**Comment by mholo65** [comment]
@devlead The following changes should now fix the problem with casing on Linux 
https://github.com/cake-build/cake/blob/e14364ed5e56eff4dac90f5c795824ad29e79dc5/src/Cake.NuGet/Install/NuGetPackageInstaller.cs#L193-L197

**Comment by devlead** [comment]
@mholo65 your changes have been merged, thanks for your contribution 👍

**Pull request merged**